app/Http/Controllers/TradeController.php

```diff
@@ -15,7 +16,28 @@ class TradeController extends Controller
      */
     public function index(Request $request)
     {
-        $trades = Trade::all();
+        $trades = Trade::where('status', 'open')
+                    ->has('items')->with('items')
+                    ->filtered($request->get('requested_items'))
+                    ->filtered($request->get('description'));
```

Comment:
So the idea with the `Filterable` trait, is that it takes a query param and searches on all the fields at the same time so what i'd recommend here is just having 1 search parameter like this
```suggestion
                    ->filtered($request->get('search'))
```